### DISCUSSION TITLE
offset error

### DISCUSSION THREAD

**Discussion by serdtsekol1** [MrThearMan/graphene-django-query-optimizer#103] [Q&A] [answered]
How can I get a zero element if when using offset 0 the error “Argument `offset` must be a positive integer.”

If I delete the code that is marked in green, as shown in the photo (1), in the original file (validators.py) of the module, then everything starts working as it should. And then I have a question, why can't I use the offset 0?
Photo1
![image](https://github.com/MrThearMan/graphene-django-query-optimizer/assets/15886730/140ffe87-94e4-4b6a-9e42-167c9baa5382)

For example, the original code returns the following results:
![image](https://github.com/MrThearMan/graphene-django-query-optimizer/assets/15886730/f1b99d2b-0008-4434-8fc8-6cc8dbbbb455)
![image](https://github.com/MrThearMan/graphene-django-query-optimizer/assets/15886730/6efc7e35-abaa-4da5-9770-54f48b41f6cf)
![image](https://github.com/MrThearMan/graphene-django-query-optimizer/assets/15886730/daaf3e1b-2575-4428-8066-72dc28ae7d42)

As you can see in the screenshots. If I use offset, I lose access to the first product because I can't use offset 0. And when I use offset 1, I get the list starts from the second product, not from the first. :(

**Comment by MrThearMan** [comment]
I made that choice in order to simplify logic related to offsets. I convert offsets to an `after` cursor value, and so a zero offset could end up pointing to a node that does not exist (i.e., "after -1 node").

While simply omitting the offset is the same as having a 0 offset, I guess I could simply ignore the offset instead of throwing an error if 0 is given. Would that be a good solution?

**Reply by serdtsekol1** [reply]
Yes, it would be convenient. Thank you)

**Reply by MrThearMan** [reply, marked as answer]
Ok, added this in [0.5.3](https://github.com/MrThearMan/graphene-django-query-optimizer/releases/tag/v0.5.3).
